src/functions/module_lookup.ts

```diff
@@ -858,3 +860,122 @@ export const PIVOT = {
   },
   returns: ["RANGE<ANY>"],
 } satisfies AddFunctionDescription;
+
+//--------------------------------------------------------------------------
+// OFFSET
+//--------------------------------------------------------------------------
+
+export const OFFSET = {
+  description: _t(
+    "Returns a range reference shifted a specified number of rows and columns from a starting cell reference."
+  ),
+  args: [
+    arg(
+      "cell_reference (meta)",
+      _t("The starting point from which to count the offset rows and columns.")
+    ),
+    arg("offset_rows (number)", _t("The number of rows to offset by.")),
+    arg("offset_columns (number)", _t("The number of columns to offset by.")),
+    arg(
+      "height (number, default='height of cell_reference')",
+      _t("The height of the range to return starting at the offset target.")
+    ),
+    arg(
+      "width (number, default='width of cell_reference')",
+      _t("The width of the range to return starting at the offset target.")
+    ),
+  ],
+  compute: function (
+    cellReference: Maybe<{ value: string }>,
+    offsetRows: Maybe<FPayload>,
+    offsetColumns: Maybe<FPayload>,
+    height: Maybe<FPayload>,
+    width: Maybe<FPayload>
+  ): Matrix<FPayload> {
+    if (isEvaluationError(cellReference?.value)) {
+      throw cellReference;
+    }
+
+    const _cellReference = cellReference?.value;
+    const zone = toZone(_cellReference!);
+
+    let offsetHeight = zone.bottom - zone.top;
+    let offsetWidth = zone.right - zone.left;
+
+    if (height) {
+      const _height = toNumber(height, this.locale);
+      assertPositive(
+        _t(`Height value is ${_height}. It should be greater than or equal to 1.`),
+        _height
+      );
+      offsetHeight = _height - 1;
+    }
+
+    if (width) {
+      const _width = toNumber(width, this.locale);
+      assertPositive(
+        _t(`width value is ${_width}. It should be greater than or equal to 1.`),
+        _width
+      );
+      offsetWidth = _width - 1;
+    }
+
+    let sheetId = this.getters.getActiveSheetId();
+
+    const { sheetName } = splitReference(_cellReference!);
+
+    if (sheetName) {
+      sheetId = this.getters.getSheetIdByName(sheetName)!; // Bad sheet name is handled by isEvaluationError() function above
```

Comment:
```suggestion
      sheetId = this.getters.getSheetIdByName(sheetName)!; // invalid references are caught upstream, we are guaranteed that the sheetId is found at this point.
```
Better to give the functional reason why we can make this assumption